iothub/service/src/IotHubCredential.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Common;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// The properties required for authentication to IoT hub that are independent of the authentication type.
+    /// </summary>
+    internal abstract class IotHubCredential
+        : IAuthorizationHeaderProvider, ICbsTokenProvider
+    {
+        private const string HostNameSeparator = ".";
+
+        public string IotHubName { get; protected set; }
+
+        public string HostName { get; protected set; }
+
+        public Uri HttpsEndpoint { get; protected set; }
+
+        public Uri AmqpEndpoint { get; protected set; }
+
+        public abstract string GetAuthorizationHeader();
+
+        public abstract Task<CbsToken> GetTokenAsync(Uri namespaceAddress, string appliesTo, string[] requiredClaims);
+
+        // Azure.Core (used in IotHubTokenCredential) is not available in NET451.
```

Comment:
So does that mean NET451 users won't be able to do RBAC auth? We may want to pass this along to our PMs in case our customers would like NET451 support

Reply:
Nope they won't be. Good call. I'll bring it up with them tomorrow.